python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -202,6 +207,32 @@ cdef class GroupBy:
 
         return Table(data=result_data, index=grouped_keys)
 
+    def replace_nulls(self, Table values, object method):
+        cdef table_view val_view = values.view()
```

Comment:
Still scratching my head to make sure but I think this might unnecessarily materialize a `RangeIndex`.

Reply:
It might, but it depends on whether the upstream passes in a `RangeIndex` in `values`. Here the meaning of `values` are the value columns in the `groupby.replace_nulls` operation, `RangeIndex` shouldn't exist. (Python interface is yet to be added so it is a bit hard to see).